packages/framework/src/resources/workflow/workflow.resource.ts

```diff
@@ -0,0 +1,182 @@
+    if (!process.env.NOVU_SECRET_KEY) {
+      throw new MissingSecretKeyError();
+    }
+
+    const unvalidatedData = (event.payload || {}) as T_PayloadUnvalidated;
+    let validatedData: T_PayloadValidated;
+    if (options.payloadSchema) {
+      const validationResult = await validateData(options.payloadSchema, unvalidatedData);
+      if (validationResult.success === false) {
+        throw new WorkflowPayloadInvalidError(workflowId, validationResult.errors);
+      }
+      validatedData = validationResult.data;
+    } else {
+      // This type coercion provides support to trigger Workflows without a payload schema
+      validatedData = event.payload as unknown as T_PayloadValidated;
+    }
+    const bridgeUrl = await getBridgeUrl();
+
+    const requestPayload = {
+      name: workflowId,
+      to: event.to,
+      payload: {
+        ...validatedData,
+      },
+      ...(event.transactionId && { transactionId: event.transactionId }),
+      ...(event.overrides && { overrides: event.overrides }),
+      ...(event.actor && { actor: event.actor }),
+      ...(bridgeUrl && { bridgeUrl }),
+    };
+
+    const result = await apiClient.post<EventTriggerResponse>('/events/trigger', requestPayload);
+
+    const cancel = async () => {
+      return apiClient.delete<CancelEventTriggerResponse>(`/events/trigger/${result.transactionId}`);
+    };
+
+    return {
+      cancel,
+      data: result,
+    };
+  };
+
+  const newWorkflow: DiscoverWorkflowOutput = {
+    workflowId,
+    options: {
+      ...options,
+      /*
+       * TODO: Transformation added for backwards compatibility, remove this additional transform after we
+       * start using `data.schema` and `control.schema` in UI.
+       */
+      inputSchema: transformSchema(options.controlSchema || options.inputSchema || emptySchema),
+      controlSchema: transformSchema(options.controlSchema || options.inputSchema || emptySchema),
+      payloadSchema: transformSchema(options.payloadSchema || emptySchema),
+    },
+    steps: [],
+    code: execute.toString(),
+    /** @deprecated */
+    data: {
+      schema: transformSchema(options.payloadSchema || emptySchema),
+      unknownSchema: options.payloadSchema || emptySchema,
+    },
+    payload: {
+      schema: transformSchema(options.payloadSchema || emptySchema),
+      unknownSchema: options.payloadSchema || emptySchema,
+    },
+    /** @deprecated */
+    inputs: {
+      schema: transformSchema(options.controlSchema || options.inputSchema || emptySchema),
+      unknownSchema: options.controlSchema || options.inputSchema || emptySchema,
+    },
+    controls: {
+      schema: transformSchema(options.controlSchema || options.inputSchema || emptySchema),
+      unknownSchema: options.controlSchema || options.inputSchema || emptySchema,
+    },
+    tags: options.tags || [],
+    preferences: mapPreferences(options.preferences),
+    name: options.name,
+    description: options.description,
```

Comment:
This file and the previous are showing a bad diff. Here's what happened:
1. renamed from `index.ts` -> `workflow.resource.ts`
2. added back `index.ts` to preserve existing imports
3. added these 2 lines